pkg/apis/catalog/basic_view.go

```diff
@@ -24,6 +25,14 @@ func (r *CreateRequest) Validate() error {
 		return err
 	}
 
+	if err := validation.IsDNSLabel(r.Name); err != nil {
+		return fmt.Errorf("invalid name: %w", err)
+	}
+
+	if len(r.Name) > 30 {
+		return fmt.Errorf("name %q is too long, must be less than 30 characters", r.Name)
+	}
```

Comment:
What happen when it exceeds the length?

Reply:
When catalog name is  too long, it is not easy for users to choose template to create resource or check info of templates. The name is not related to resource deployment, we don't need to worry about it.

Reply:
Let's handle the name limit in other Pull Request.